src/main/java/io/vertx/proton/ProtonReceiver.java

```diff
@@ -35,6 +35,27 @@
    */
   ProtonReceiver handler(ProtonMessageHandler handler);
 
+  /**
+   * Sets the handler to notify about a message from the peer that exceeds this receiver's configured
+   * maximum message size.
+   * <p>
+   * The handler will be invoked as soon as a transfer frame from the peer has been received which
+   * causes the max-message-size to be exceeded. This means that the full message may not have been
+   * received by that time.
+   * <p>
+   * The AMQP 1.0 specification requires the link to be detached with error code
+   * <em>amqp:link:message-size-exceeded</em> in such a situation. Thus, after the handler returns,
+   * a corresponding AMQP <em>detach</em> frame with <em>close=true</em> will be sent to the peer,
```

Comment:
I think this should be a non-closing detach (or it could inspect the terminus durability and choose accordingly). In many cases the resulting effect will be exactly the same, but using closed=true can e.g mean removal of a durable subscription which isnt something I think this should cause. If someone wants to send a closing detach they can use the callback to do it.